hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -439,7 +439,7 @@ public boolean containsInstant(String ts) {
     // Check for older timestamp which have sec granularity and an extension of DEFAULT_MILLIS_EXT may have been added via Timeline operations
     if (ts.length() == HoodieInstantTimeGenerator.MILLIS_INSTANT_TIMESTAMP_FORMAT_LENGTH && ts.endsWith(HoodieInstantTimeGenerator.DEFAULT_MILLIS_EXT)) {
       final String actualOlderFormatTs = ts.substring(0, ts.length() - HoodieInstantTimeGenerator.DEFAULT_MILLIS_EXT.length());
-      return containsOrBeforeTimelineStarts(actualOlderFormatTs);
+      return containsInstant(actualOlderFormatTs);
```

Comment:
The start instant comparison is neede because query of all historical instants is costly, the comparison is an expedience.

Reply:
> The start instant comparison is neede because query of all historical instants is costly, the comparison is an expedience.

When worked in inflight timeline , it cause incorrect result . Instant which's timestamp is end with default_mills_ext don't contain in inflight timeline but return true.

Reply:
I think you are correct, this is a compatibility fix, if an instant is archived, the inflight timeline would deem it as included which is not as expected. Can you check that the test `TestHoodieIndex.testCheckIfValidCommit` which is introduced together with the fix still works?

Reply:
> I think you are correct, this is a compatibility fix, if an instant is archived, the inflight timeline would deem it as included which is not as expected. Can you check that the test `TestHoodieIndex.testCheckIfValidCommit` which is introduced together with the fix still works?

`TestHoodieIndex.testCheckIfValidCommit` apply `HoodieIndexUtils.checkIfValidCommit` which include containsOrBeforeTimelineStarts but not containsInstant. I new a inflight instant check with containsInstant in unit test and unit test is passed.